### PR TITLE
Dont repeat full object names where possible

### PR DESCRIPTION
**Pull request by mbrevda**
A tiny size gainz for PropTypes

### CODE REVIEW

File: modules/PropTypes.js [hunk 1]
```diff
@@ -1,48 +1,50 @@
 import { PropTypes } from 'react'
+const {func, string, any, oneOf, shape} = PropTypes
 
-export const action = PropTypes.oneOf([
+const FuncReq = func.isRequired
+const StrReq = StrReq
```
Comment (jochenberger) [inline modules/PropTypes.js]: shouldn't this read `const StrReq =  string.isRequired`?

Comment (mbrevda) [inline modules/PropTypes.js]: search & replace. Fixed now.

File: modules/PropTypes.js [hunk 2]
```diff
@@ -1,48 +1,50 @@
 import { PropTypes } from 'react'
+const {func, string, any, oneOf, shape} = PropTypes
```
Comment (timdorr) [inline modules/PropTypes.js]: Nit: Add spaces inside of the brackets. That's consistent with how we normally write destructurings.

Comment (ryanflorence) [inline modules/PropTypes.js]: if we're gonna nit on this stuff I suppose we should put it in the eslint config.

**Comment by ryanflorence** [comment]
just curious, what's the size change?

File: modules/PropTypes.js (hunk 2 above)
Comment (timdorr) [inline modules/PropTypes.js]: Yeah, it's come up a couple times before. I can get that added.

**Comment by mbrevda** [comment]
@ryanflorence embarrassingly small (like .5k or so). This was more as a sodlace when I realized that #4046 wasn't going to happen

**Comment by alisd23** [comment]
Is the .5k really worth it? Seems like it's complicating the PropTypes file for a really tiny gain.

**Comment by alisd23** [comment]
I can understand destructing the PropTypes variable, but is the func.isRequired -> funqReq really necessary? After minification aren't we looking at a difference of 10's of *characters*?

**Comment by mbrevda** [comment]
> what's the size change?

I've confirmed that my bundle size decreases by .5k

> Is the .5k really worth it?

I think that's subjective. Personally, I'd love to drop every single byte that I can.

> is the func.isRequired -> funqReq really necessary

I appreciate not minifying code at the expense of readability. I think that propType-like variables are pretty obvious to most. I wouldn't necessarily rename methods or classnames on the basis of minification, and the exported values are quite self documenting.

I do agree that a babel-transform to do this work at build time would be preferable. I do not know of any that can do that for us though

**Comment by ryanflorence** [comment]
I'd rather not worry about micro stuff like this right now, trying to ship this and get it right first.

**Pull request closed without merging**